### ISSUE TITLE
Memory symbols inconsistencies

### ISSUE THREAD

**Issue by tilleul** [AppleWin/AppleWin#1145]
It looks like the two .SYM files are completely disorganised:

- Many symbols are missing, even in zero page. 
- Namings are inconsistent: for instance we have MON.A1L, MON.A1H, MON.A2L, MON.A2H but then we have A3L, A3H, A4L, A4H, A5L, A5H.
- Some symbols are overwritten within the same file: for instance A2C.ROMSTATE and TEXT.HOLE.1
- Several names are not official and/or too long: for instance MOVE.HIGHEST.STRING.TO.TOP ($E562) should be GRBGPASS 
- Symbols are unordered making it hard to search/update/fix anything in these files

I'm ready to start dusting off these files if this can help but are there any guidelines for the naming of symbols and for the internal and general organisation of these files ?
What about memory locations that have different usage realms (like monitor, applesoft, integer, dos, prodos, ][+ ROM, //e ROM, //c only, etc) ? Shouldn't they be in their own file ?
Should the symbols be distributed in more files ? Maybe one per version of computer (for instance there are -- of course -- many differences between the ][+ and the //e ROMs -- having //e symbols loaded while emulating a ][+ seems inappropriate) ? Or maybe one per usage realm (applesoft, monitor, dos, etc) ?
What about symbols that span many bytes like FAC, RESULT or TEMPST ? Should each byte be named ?
What are the ".01=addr .02=addr" etc in the A2_BASIC.SYM file ?
Are there any guidelines for documenting where the names came from as some might be rather obscure ?

**Comment by tomcw** [comment]
Hi @Michaelangel007 - any comment on this issue?

Probably specific .SYM files for different machines' ROMs (eg. II, II+, //e, enhanced //e, ...and then clones?).

And .SYM files for each card's firmware (eg. Disk II, HDD cards)... adapting to the current slot/card configuration, and prefixed with slot number so multiple slots with the same card are supported (eg. Disk II in slots 5 and 6).

It all sounds like a massive job! What would be the priority?

**Comment by Michaelangel007** [comment]
Yeah, the symbol table is a real mess. :-/  One of the big problem is that it only describes labels and not data types which is the reason I developed `DB`, `DF`, etc. so the debugger can display more complicated types such as Applesoft's FAC, etc. natively.  

We probably should switch over to a the improved symbol table file `.sym2` and keep the regular `.sym` files as the "dumb" `ADDRESS SYMBOL` format.

👍  for symbol tables specific to each machine type.

To manage this complexity I recommend we break the memory space up into sections.  Namely:

* Zero-Page
* Text Screen
* I/O $C000-$CFFF
* Applesoft $D000-$F7FF
* ROM $F800-$FFFF

That way we can slowly upgrade the symbols.

Having official source listings would be preferred. 

I'm also a fan of long, descriptive symbol names [COPY.FAC.TO.ARG.ROUNDED](http://www.txbobsc.com/scsc/scdocumentor/E913.html) to minimize naming collisions which is why we see warnings such as `COPY.FAC.TO.ARG.ROUNDED (23 > 20)`.

For Applesoft I'm also big fan off Bob Sander-Cederlof's [unofficial Applesoft names](http://www.txbobsc.com/scsc/scdocumentor/) since they are more well known but I'm also OK with using the [official source variables](https://github.com/cmosher01/Apple-II-Source/blob/master/src/system/applesoft/applesoft.m4).

I am OK with prefixing all the $F800-$FFFF ROM symbols as `F8.HOME`

If people have suggestions I'm open to hearing them to simplify managing this mess.

**Comment by Michaelangel007** [comment]
> What are the ".01=addr .02=addr" etc in the A2_BASIC.SYM file ? 

Those are **private entry points.**

```
D56C PARSE
	  .01=D56D  .02=D578  .03=D588  .04=D590  .05=D5A2  .06=D5A7  .07=D5A8
	  .08=D5CB  .09=D5CD  .10=D5E0  .11=D5E2  .12=D5E9  .13=D5F2  .14=D5F9
	  .15=D5FD  .16=D604  .17=D610
```

Looking at Bob's source for [D365](http://www.txbobsc.com/scsc/scdocumentor/D52C.html)

```
               1270 *--------------------------------
               1280 *      TOKENIZE THE INPUT LINE
               1290 *--------------------------------
               1300 PARSE.INPUT.LINE
D559- A6 B8    1310        LDX TXTPTR   INDEX INTO UNPARSED LINE
D55B- CA       1320        DEX          PREPARE FOR INX AT "PARSE"
D55C- A0 04    1330        LDY #4       INDEX TO PARSED OUTPUT LINE
D55E- 84 13    1340        STY DATAFLG  CLEAR SIGN-BIT OF DATAFLG
D560- 24 D6    1350        BIT LOCK     IS THIS PROGRAM LOCKED?
D562- 10 08    1360        BPL PARSE    NO, GO AHEAD AND PARSE THE LINE
D564- 68       1370        PLA          YES, IGNORE INPUT AND "RUN"
D565- 68       1380        PLA             THE PROGRAM
D566- 20 65 D6 1390        JSR SETPTRS  CLEAR ALL VARIABLES
D569- 4C D2 D7 1400        JMP NEWSTT   START RUNNING
               1410 *--------------------------------
D56C- E8       1420 PARSE  INX          NEXT INPUT CHARACTER
D56D- BD 00 02 1430 .1     LDA INPUT.BUFFER,X
D570- 24 13    1440        BIT DATAFLG       IN A "DATA" STATEMENT?
D572- 70 04    1450        BVS .2            YES (DATAFLG = $49)
D574- C9 20    1460        CMP #' '     IGNORE BLANKS
D576- F0 F4    1470        BEQ PARSE
D578- 85 0E    1480 .2     STA ENDCHR
D57A- C9 22    1490        CMP #'"      START OF QUOTATION?
D57C- F0 74    1500        BEQ .13
D57E- 70 4D    1510        BVS .9       BRANCH IF IN "DATA" STATEMENT
D580- C9 3F    1520        CMP #'?      SHORTHAND FOR "PRINT"?
D582- D0 04    1530        BNE .3       NO
D584- A9 BA    1540        LDA #TOKEN.PRINT  YES, REPLACE WITH "PRINT" TOKEN
D586- D0 45    1550        BNE .9       ...ALWAYS
D588- C9 30    1560 .3     CMP #'0      IS IT A DIGIT, COLON, OR SEMI-COLON?
D58A- 90 04    1570        BCC .4       NO, PUNCTUATION !"#$%&'()*+,-./
D58C- C9 3C    1580        CMP #';'+1
D58E- 90 3D    1590        BCC .9       YES, NOT A TOKEN
```

**Comment by Michaelangel007** [comment]
> What about symbols that span many bytes like FAC, RESULT or TEMPST ?

That's what `DB`, `DA`, `ASC`, `DF` are for.  (DB takes a byte range.)
